internal/pkg/manifest/workload.go

```diff
@@ -66,6 +74,40 @@ type Image struct {
 	DependsOn    map[string]string `yaml:"depends_on,flow"` // Add any sidecar dependencies.
 }
 
+type imageTransformer struct{}
+
+// Transformer implements customized merge logic for Image field of manifest.
+// It merges `DockerLabels` and `DependsOn` in the default manager (i.e. with configurations mergo.WithOverride, mergo.WithOverwriteWithEmptyValue)
+// And then overrides both `Build` and `Location` fields at the same time with the src values, given that they are non-empty themselves.
+func (t imageTransformer) Transformer(typ reflect.Type) func(dst, src reflect.Value) error {
+	if typ != reflect.TypeOf(Image{}) {
+		return nil
+	}
+	return func(dst, src reflect.Value) error {
+		// Perform default merge
+		dstImage := dst.Interface().(Image)
+		srcImage := src.Interface().(Image)
```

Comment:
Do we need to worry about panicking here? Should we be using `CanInterface` first?

Reply:
It will panic if `if the Value was obtained by accessing unexported struct fields`, while `mergo` only merges exported fields, so I think we don't need to worry about panic here.

I also did some experiment where I added an unexported field to `LoadBalancedWebService` (the receiver of `ApplyEnv`). The unexported field won't get merged, hence `Transformer` won't get called.